### DISCUSSION TITLE
handling /index.tsx and /[id].tsx behaviour with react navigation

### DISCUSSION THREAD

**Discussion by louisholley** [nandorojo/solito#101] [Q&A] [answered]
we have the following routes in our app:
- `/community/questions` - this page is a list of all the questions
- `/community/questions/[id]` - this is a specific question with all of its answers

this sort of pattern works fine with nextjs usually, but it feels like we have to do a bit of a workaround in order to achieve consistent routing with solito, where both `community/questions/index.tsx` and `community/questions/[id].tsx` export a page like this:

```
const { useParam } = createParam<{ id: string }>();

export const CommunityQuestionsScreen = () => {
  const [questionId] = useParam("id");
  if (questionId) return <CommunityQuestionScreen questionId={questionId} />;
  return (
    <Screen>
      <H1>Community questions screen</H1>
    </Screen>
  );
};
```

is this the recommended way to handle this pattern, or is there a nicer way that I'm not thinking of? cheers!

**Comment by nandorojo** [comment, marked as answer]
not sure I understand exactly…why do you need to do a work around? the normal behavior works fine. have you seen the `user-detail` in the example app?

**Reply by louisholley** [reply]
so imagine the `user-detail` route is renamed to `users`, such that you can visit `/users/123` to see the details of the user with id = 123. but imagine also, you want the route `/users` to render a list of all the users. this is our use case - you can achieve this in nextjs with an `index.tsx` route as well as a `[param].tsx` route in the same dir, but I can't currently see how this is possible with solito (react navigation requires you to map `/users/:id` to a specific component, and i don't think you'd be able to map `/users` to a different component). please let me know if you have any ideas!

mainly the problem is in [this file](https://github.com/nandorojo/solito/blob/master/example-monorepos/blank/packages/app/provider/navigation/index.tsx) i think

**Reply by nandorojo** [reply]
you can just do this:

```ts
screens: {
 home: '',
 'user-detail': 'users/:id',
 'user-list': 'users'
}
```

**Reply by louisholley** [reply]
right you are! my bad lol, thanks so much

**Reply by nandorojo** [reply]
cool! np
